### PR TITLE
[debug dump util] FDB debug dump util changes

### PR DESCRIPTION
**Pull request by dgsudharsan**
Signed-off-by: Sudharsan Dhamal Gopalarathnam sudharsand@nvidia.com

#### What I did
Added FDB debug dump module

#### How I did it
Used debug dump infra to add the FDB debug dump utility

#### How to verify it
Added Unit test cases to verify it.

#### New command output (if the output of a command-line utility has changed)

```
root@sonic:~# dump state fdb all
{
    "Vlan50:04:3f:72:e3:70:08": {
        "APPL_DB": {
            "keys": [],
            "tables_not_found": []
        },
        "ASIC_DB": {
            "keys": [
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:oid:0x3a000000000d23": {
                        "SAI_BRIDGE_PORT_ATTR_ADMIN_STATE": "true",
                        "SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE": "SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW",
                        "SAI_BRIDGE_PORT_ATTR_PORT_ID": "oid:0x100000000093e",
                        "SAI_BRIDGE_PORT_ATTR_TYPE": "SAI_BRIDGE_PORT_TYPE_PORT"
                    }
                },
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:{\"bvid\":\"oid:0x26000000000d22\",\"mac\":\"04:3F:72:E3:70:08\",\"switch_id\":\"oid:0x21000000000000\"}": {
                        "SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID": "oid:0x3a000000000d23",
                        "SAI_FDB_ENTRY_ATTR_PACKET_ACTION": "SAI_PACKET_ACTION_FORWARD",
                        "SAI_FDB_ENTRY_ATTR_TYPE": "SAI_FDB_ENTRY_TYPE_DYNAMIC"
                    }
                }
            ],
            "tables_not_found": [],
            "vidtorid": {
                "oid:0x3a000000000d23": "oid:0x20000003a"
            }
        },
        "STATE_DB": {
            "keys": [
                {
                    "FDB_TABLE|Vlan50:04:3f:72:e3:70:08": {
                        "port": "Ethernet0",
                        "type": "dynamic"
                    }
                }
            ],
            "tables_not_found": []
        }
    },
    "Vlan10:04:3f:72:ce:80:8b": {
        "APPL_DB": {
            "keys": [],
            "tables_not_found": []
        },
        "ASIC_DB": {
            "keys": [
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:oid:0x3a000000000d18": {
                        "SAI_BRIDGE_PORT_ATTR_ADMIN_STATE": "true",
                        "SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE": "SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW",
                        "SAI_BRIDGE_PORT_ATTR_PORT_ID": "oid:0x100000000053f",
                        "SAI_BRIDGE_PORT_ATTR_TYPE": "SAI_BRIDGE_PORT_TYPE_PORT"
                    }
                },
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:{\"bvid\":\"oid:0x26000000000d20\",\"mac\":\"04:3F:72:CE:80:8B\",\"switch_id\":\"oid:0x21000000000000\"}": {
                        "SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID": "oid:0x3a000000000d18",
                        "SAI_FDB_ENTRY_ATTR_PACKET_ACTION": "SAI_PACKET_ACTION_FORWARD",
                        "SAI_FDB_ENTRY_ATTR_TYPE": "SAI_FDB_ENTRY_TYPE_DYNAMIC"
                    }
                }
            ],
            "tables_not_found": [],
            "vidtorid": {
                "oid:0x3a000000000d18": "oid:0x10000003a"
            }
        },
        "STATE_DB": {
            "keys": [
                {
                    "FDB_TABLE|Vlan10:04:3f:72:ce:80:8b": {
                        "port": "Ethernet252",
                        "type": "dynamic"
                    }
                }
            ],
            "tables_not_found": []
        }
    },
    "Vlan690:04:3f:72:ce:80:8b": {
        "APPL_DB": {
            "keys": [],
            "tables_not_found": []
        },
        "ASIC_DB": {
            "keys": [
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:oid:0x3a000000000d18": {
                        "SAI_BRIDGE_PORT_ATTR_ADMIN_STATE": "true",
                        "SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE": "SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW",
                        "SAI_BRIDGE_PORT_ATTR_PORT_ID": "oid:0x100000000053f",
                        "SAI_BRIDGE_PORT_ATTR_TYPE": "SAI_BRIDGE_PORT_TYPE_PORT"
                    }
                },
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:{\"bvid\":\"oid:0x26000000000d1c\",\"mac\":\"04:3F:72:CE:80:8B\",\"switch_id\":\"oid:0x21000000000000\"}": {
                        "SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID": "oid:0x3a000000000d18",
                        "SAI_FDB_ENTRY_ATTR_PACKET_ACTION": "SAI_PACKET_ACTION_FORWARD",
                        "SAI_FDB_ENTRY_ATTR_TYPE": "SAI_FDB_ENTRY_TYPE_DYNAMIC"
                    }
                }
            ],
            "tables_not_found": [],
            "vidtorid": {
                "oid:0x3a000000000d18": "oid:0x10000003a"
            }
        },
        "STATE_DB": {
            "keys": [
                {
                    "FDB_TABLE|Vlan690:04:3f:72:ce:80:8b": {
                        "port": "Ethernet252",
                        "type": "dynamic"
                    }
                }
            ],
            "tables_not_found": []
        }
    },
    "Vlan40:04:3f:72:e3:70:09": {
        "APPL_DB": {
            "keys": [],
            "tables_not_found": []
        },
        "ASIC_DB": {
            "keys": [
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:oid:0x3a000000000d16": {
                        "SAI_BRIDGE_PORT_ATTR_ADMIN_STATE": "true",
                        "SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE": "SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW",
                        "SAI_BRIDGE_PORT_ATTR_PORT_ID": "oid:0x2000000000d13",
                        "SAI_BRIDGE_PORT_ATTR_TYPE": "SAI_BRIDGE_PORT_TYPE_PORT"
                    }
                },
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:{\"bvid\":\"oid:0x26000000000d15\",\"mac\":\"04:3F:72:E3:70:09\",\"switch_id\":\"oid:0x21000000000000\"}": {
                        "SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID": "oid:0x3a000000000d16",
                        "SAI_FDB_ENTRY_ATTR_PACKET_ACTION": "SAI_PACKET_ACTION_FORWARD",
                        "SAI_FDB_ENTRY_ATTR_TYPE": "SAI_FDB_ENTRY_TYPE_DYNAMIC"
                    }
                }
            ],
            "tables_not_found": [],
            "vidtorid": {
                "oid:0x3a000000000d16": "oid:0x3a"
            }
        },
        "STATE_DB": {
            "keys": [
                {
                    "FDB_TABLE|Vlan40:04:3f:72:e3:70:09": {
                        "port": "PortChannel0002",
                        "type": "dynamic"
                    }
                }
            ],
            "tables_not_found": []
        }
    },
    "Vlan691:04:3f:72:ce:80:8b": {
        "APPL_DB": {
            "keys": [],
            "tables_not_found": []
        },
        "ASIC_DB": {
            "keys": [
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:oid:0x3a000000000d18": {
                        "SAI_BRIDGE_PORT_ATTR_ADMIN_STATE": "true",
                        "SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE": "SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW",
                        "SAI_BRIDGE_PORT_ATTR_PORT_ID": "oid:0x100000000053f",
                        "SAI_BRIDGE_PORT_ATTR_TYPE": "SAI_BRIDGE_PORT_TYPE_PORT"
                    }
                },
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:{\"bvid\":\"oid:0x26000000000d1e\",\"mac\":\"04:3F:72:CE:80:8B\",\"switch_id\":\"oid:0x21000000000000\"}": {
                        "SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID": "oid:0x3a000000000d18",
                        "SAI_FDB_ENTRY_ATTR_PACKET_ACTION": "SAI_PACKET_ACTION_FORWARD",
                        "SAI_FDB_ENTRY_ATTR_TYPE": "SAI_FDB_ENTRY_TYPE_DYNAMIC"
                    }
                }
            ],
            "tables_not_found": [],
            "vidtorid": {
                "oid:0x3a000000000d18": "oid:0x10000003a"
            }
        },
        "STATE_DB": {
            "keys": [
                {
                    "FDB_TABLE|Vlan691:04:3f:72:ce:80:8b": {
                        "port": "Ethernet252",
                        "type": "dynamic"
                    }
                }
            ],
            "tables_not_found": []
        }
    },
    "Vlan69:04:3f:72:e3:70:09": {
        "APPL_DB": {
            "keys": [],
            "tables_not_found": []
        },
        "ASIC_DB": {
            "keys": [
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:oid:0x3a000000000d16": {
                        "SAI_BRIDGE_PORT_ATTR_ADMIN_STATE": "true",
                        "SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE": "SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW",
                        "SAI_BRIDGE_PORT_ATTR_PORT_ID": "oid:0x2000000000d13",
                        "SAI_BRIDGE_PORT_ATTR_TYPE": "SAI_BRIDGE_PORT_TYPE_PORT"
                    }
                },
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:{\"bvid\":\"oid:0x26000000000d1a\",\"mac\":\"04:3F:72:E3:70:09\",\"switch_id\":\"oid:0x21000000000000\"}": {
                        "SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID": "oid:0x3a000000000d16",
                        "SAI_FDB_ENTRY_ATTR_PACKET_ACTION": "SAI_PACKET_ACTION_FORWARD",
                        "SAI_FDB_ENTRY_ATTR_TYPE": "SAI_FDB_ENTRY_TYPE_DYNAMIC"
                    }
                }
            ],
            "tables_not_found": [],
            "vidtorid": {
                "oid:0x3a000000000d16": "oid:0x3a"
            }
        },
        "STATE_DB": {
            "keys": [
                {
                    "FDB_TABLE|Vlan69:04:3f:72:e3:70:09": {
                        "port": "PortChannel0002",
                        "type": "dynamic"
                    }
                }
            ],
            "tables_not_found": []
        }
    },
    "Vlan101:04:3f:72:e3:70:09": {
        "APPL_DB": {
            "keys": [],
            "tables_not_found": []
        },
        "ASIC_DB": {
            "keys": [
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:oid:0x3a000000000d16": {
                        "SAI_BRIDGE_PORT_ATTR_ADMIN_STATE": "true",
                        "SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE": "SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW",
                        "SAI_BRIDGE_PORT_ATTR_PORT_ID": "oid:0x2000000000d13",
                        "SAI_BRIDGE_PORT_ATTR_TYPE": "SAI_BRIDGE_PORT_TYPE_PORT"
                    }
                },
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:{\"bvid\":\"oid:0x26000000000d28\",\"mac\":\"04:3F:72:E3:70:09\",\"switch_id\":\"oid:0x21000000000000\"}": {
                        "SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID": "oid:0x3a000000000d16",
                        "SAI_FDB_ENTRY_ATTR_PACKET_ACTION": "SAI_PACKET_ACTION_FORWARD",
                        "SAI_FDB_ENTRY_ATTR_TYPE": "SAI_FDB_ENTRY_TYPE_DYNAMIC"
                    }
                }
            ],
            "tables_not_found": [],
            "vidtorid": {
                "oid:0x3a000000000d16": "oid:0x3a"
            }
        },
        "STATE_DB": {
            "keys": [
                {
                    "FDB_TABLE|Vlan101:04:3f:72:e3:70:09": {
                        "port": "PortChannel0002",
                        "type": "dynamic"
                    }
                }
            ],
            "tables_not_found": []
        }
    },
    "Vlan40:04:3f:72:ce:80:8b": {
        "APPL_DB": {
            "keys": [],
            "tables_not_found": []
        },
        "ASIC_DB": {
            "keys": [
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:oid:0x3a000000000d18": {
                        "SAI_BRIDGE_PORT_ATTR_ADMIN_STATE": "true",
                        "SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE": "SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW",
                        "SAI_BRIDGE_PORT_ATTR_PORT_ID": "oid:0x100000000053f",
                        "SAI_BRIDGE_PORT_ATTR_TYPE": "SAI_BRIDGE_PORT_TYPE_PORT"
                    }
                },
                {
                    "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:{\"bvid\":\"oid:0x26000000000d15\",\"mac\":\"04:3F:72:CE:80:8B\",\"switch_id\":\"oid:0x21000000000000\"}": {
                        "SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID": "oid:0x3a000000000d18",
                        "SAI_FDB_ENTRY_ATTR_PACKET_ACTION": "SAI_PACKET_ACTION_FORWARD",
                        "SAI_FDB_ENTRY_ATTR_TYPE": "SAI_FDB_ENTRY_TYPE_DYNAMIC"
                    }
                }
            ],
            "tables_not_found": [],
            "vidtorid": {
                "oid:0x3a000000000d18": "oid:0x10000003a"
            }
        },
        "STATE_DB": {
            "keys": [
                {
                    "FDB_TABLE|Vlan40:04:3f:72:ce:80:8b": {
                        "port": "Ethernet252",
                        "type": "dynamic"
                    }
                }
            ],
            "tables_not_found": []
        }
    }
}
root@sonic:~#
```

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@dgsudharsan could you please review failures?

**Comment by dgsudharsan** [comment]
> @dgsudharsan could you please review failures?

Looks like an infrastructure failure and it is seen across pipeline
ERROR: Could not install packages due to an OSError: [Errno 2] No such file or directory: '/__w/1/target/python-wheels/swsssdk-2.0.1-py3-none-any.whl'


@lguohan  @qiluo-msft  Can you please help?

**Comment by qiluo-msft** [comment]
It should be fixed by https://github.com/Azure/sonic-utilities/pull/1971
Could you rebase to latest code?

**Comment by liat-grozovik** [comment]
@SuvarnaMeenakshi kindly reminder, can you please review?

File: dump/plugins/fdb.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"], False)
+        req = MatchRequest(db="APPL_DB", table="VXLAN_FDB_TABLE", key_pattern=fdb_name, ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"], False)
+        req = MatchRequest(db="APPL_DB", table="MCLAG_FDB_TABLE", key_pattern=fdb_name, ns=self.ns)
+        ret = self.match_engine.fetch(req)
+        self.add_to_ret_template(req.table, req.db, ret["keys"], ret["error"], False)
+
+    def init_asic_fdb_info(self, fdb_name):
+        key_split = fdb_name.split(":",1)
```
Comment (SuvarnaMeenakshi) [inline dump/plugins/fdb.py:47]: Will be good to check if ":" exists in fdb_name before splitting and update ret_temp accordingly.

Comment (dgsudharsan) [reply dump/plugins/fdb.py:53]: Done

**Review by SuvarnaMeenakshi — APPROVED** [review]
lgtm

**Pull request merged**